documentation/quarkus/guide-tkit.asciidoc

```diff
@@ -0,0 +1,192 @@
+:toc: macro
+toc::[]
+
+= tkit extensions
+
+1000kit (tkit) is a set of extensions and plugins for Quarkus.
+It was developed internally by Capgemini, so the source code of the extensions is not publicly available.
+
+However, you can still use the extensions in your Quarkus project.
+This guide explains some of the tkit extensions and how to use them.
+
+== tkit-quarkus-log
+
+The tkit logger extension consists of three extensions:
+
+* tkit-quarkus-log-cdi
+* tkit-quarkus-log-rs
+* tkit-quarkus-log-json
+
+=== tkit-quarkus-log-cdi
+
+The `tkit-quarkus-log-cdi` extension is the core logging library for Quarkus ArC, the CDI implementation in Quarkus.
+It allows automatic logging of method invocations with start execution time, parameters, result and the time of execution.
+
+Add the following dependency to your project to add the extension:
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.tkit.quarkus</groupId>
+  <artifactId>tkit-quarkus-log-cdi</artifactId>
+  <version>1.5.0</version>
+</dependency>
+----
+
+Set `quarkus.tkit.log.packages` in your application.properties file to specify the classes whose method calls should be logged.
+You can exclude classes by specifying an ignore pattern via the `quarkus.tkit.log.ignore.pattern` property.
+
+[source, properties]
+----
+quarkus.tkit.log.packages=com.devonfw.quarkus
+quarkus.tkit.log.ignore.pattern=.*MapperImpl
+----
+
+=== tkit-quarkus-log-rs
+
+This extension provides automatic logging of all HTTP communications, including information about the URL, the status code, the HTTP method or the time required to execute the request.
+Add it with the following dependency:
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.tkit.quarkus</groupId>
+  <artifactId>tkit-quarkus-log-rs</artifactId>
+  <version>1.5.0</version>
+</dependency>
+----
+
+=== tkit-quarkus-log-json
+
+`tkit-quarkus-log-json` provides automatic JSON formatting of log messages.
```

Comment:
The json log extension provides some additional features that standard quarkus-json logger does not have - e.g. stacktrace spliting, field relabeling, env var output, nested json objects etc but for most projects it is recommended that they use the official json logger instead, unless they need those extra features (like we do in our projects). More info in the README of the lib project